src/Sulu/Bundle/WebsiteBundle/Trash/AnalyticsTrashItemHandler.php

```diff
@@ -0,0 +1,237 @@
+        $this->setData($analytics, $data);
+
+        $this->domainEventCollector->collect(
+            new AnalyticsRestoredEvent($analytics, $data)
+        );
+
+        try {
+            $this->analyticsRepository->findById($id);
+        } catch (NoResultException $e) {
+            $this->doctrineRestoreHelper->persistAndFlushWithId($analytics, $id);
+
+            return $analytics;
+        }
+
+        $this->entityManager->persist($analytics);
+        $this->entityManager->flush();
+
+        return $analytics;
+    }
+
+    /**
+     * @see AnalyticsManager::setData()
+     *
+     * @param array<string, mixed> $data
+     */
+    private function setData(AnalyticsInterface $analytics, array $data): void
+    {
+        $analytics->setTitle($this->getValue($data, 'title'));
+        $analytics->setType($this->getValue($data, 'type'));
+        $analytics->setWebspaceKey($this->getValue($data, 'webspaceKey'));
+        $analytics->setContent($this->getValue($data, 'content', ''));
+        $analytics->setAllDomains($this->getValue($data, 'allDomains', false));
+
+        if ($analytics->isAllDomains()) {
+            $analytics->clearDomains();
+
+            return;
+        }
+
+        $domains = [];
+        $domainCollection = $analytics->getDomains();
+        if ($domainCollection) {
+            $domains = $domainCollection->toArray();
+        }
+
+        foreach ($this->getValue($data, 'domains', []) as $domain) {
+            if (\in_array($domain, $domains)) {
+                unset($domains[\array_search($domain, $domains)]);
+
+                continue;
+            }
+
+            $domainEntity = $this->findOrCreateNewDomain($domain);
+            $analytics->addDomain($domainEntity);
+        }
+
+        foreach ($domains as $domain) {
+            $domainEntity = $this->findOrCreateNewDomain($domain);
+            $analytics->removeDomain($domainEntity);
+        }
+    }
+
+    /**
+     * @see AnalyticsManager::findOrCreateNewDomain()
+     */
+    private function findOrCreateNewDomain(string $domain): Domain
+    {
+        /** @var Domain|null $domainEntity */
+        $domainEntity = $this->domainRepository->findByUrlAndEnvironment($domain, $this->environment);
+
+        if (null !== $domainEntity) {
+            return $domainEntity;
+        }
+
+        $domainEntity = new Domain();
+        $domainEntity->setUrl($domain);
+        $domainEntity->setEnvironment($this->environment);
+
+        $this->entityManager->persist($domainEntity);
+
+        return $domainEntity;
+    }
+
+    /**
+     * @see AnalyticsManager::getValue()
+     *
+     * @param array<string, mixed> $data
+     * @param mixed|null $default
+     *
+     * @return mixed
+     */
+    private function getValue(array $data, string $name, $default = null)
+    {
+        if (!\array_key_exists($name, $data)) {
+            return $default;
+        }
+
+        return $data[$name];
+    }
```

Comment:
These methods are pretty much the same as in the `AnalyticsManager` ... Should we share them somehow?